resources/hg-ReferralComposition.xml

```diff
@@ -1,556 +1,556 @@
-      <path value="Composition.section.code" />
-      <min value="1" />
-      <patternCodeableConcept>
-        <coding>
-          <system value="http://snomed.info/sct" />
-          <code value="440378000" />
-        </coding>
-      </patternCodeableConcept>
-    </element>
-    <element id="Composition.section:messageReason.entry">
-      <path value="Composition.section.entry" />
-      <slicing>
-        <discriminator>
-          <type value="value" />
-          <path value="$this" />
-        </discriminator>
-        <rules value="open" />
-      </slicing>
-    </element>
-    <element id="Composition.section:messageReason.entry:problem">
-      <path value="Composition.section.entry" />
-      <sliceName value="problem" />
-      <short value="Problem" />
-      <definition value="Een probleem beschrijft een toestand met betrekking tot de gezondheid en/of het welzijn van een individu. Deze toestand kan zijn benoemd door de patiënt zelf (een klacht), of door zijn of haar zorgverlener (onder andere een diagnose)." />
-      <alias value="Probleem" />
-      <max value="1" />
-      <type>
-        <code value="Reference" />
-        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-Problem" />
-      </type>
-      <mapping>
-        <identity value="ART-DECOR-hg-Kern" />
-        <map value="hg-dataelement-1886" />
-        <comment value="Probleem" />
-      </mapping>
-    </element>
+        <element id="Composition.section:messageReason">
+            <path value="Composition.section" />
+            <sliceName value="messageReason" />
+            <label value="RedenBericht" />
+            <short value="MessageReason" />
+            <definition value="Geeft de reden van de verwijzing of de update van een huisarts naar een paramedicus.&#xD;&#xA;Hierbij is de beschrijving als vrije tekst op aangeven van het NHG verplicht.&#xD;&#xA;Daarnaast kan er ook een ICPC-code van de episode worden meegestuurd, al dan niet aangevuld met meer details over de vastlegging van de ICPC." />
+            <min value="1" />
+            <max value="1" />
+            <mapping>
+                <identity value="ART-DECOR-hg-Kern" />
+                <map value="hg-dataelement-1872" />
+                <comment value="RedenBericht" />
+            </mapping>
+        </element>
+        <element id="Composition.section:messageReason.extension">
+            <path value="Composition.section.extension" />
+            <slicing>
+                <discriminator>
+                    <type value="value" />
+                    <path value="url" />
+                </discriminator>
+                <rules value="open" />
+            </slicing>
+            <min value="0" />
+        </element>
+        <element id="Composition.section:messageReason.extension:contextTextValue">
+            <path value="Composition.section.extension" />
+            <sliceName value="contextTextValue" />
+            <min value="0" />
```

Comment:
The minimum cardinality of this slice should be '1', in line with the functional data set.